update/update_2.3.5-2.4.php

```diff
@@ -193,6 +193,28 @@
 			if(!@mysqli_query($connid, "CREATE TABLE ".$db_settings['read_status_table']." (user_id int(11) UNSIGNED NOT NULL, posting_id int(11) UNSIGNED NOT NULL, time timestamp NOT NULL, PRIMARY KEY (user_id, posting_id)) CHARSET=utf8 COLLATE=utf8_general_ci;")) {
 				$update['errors'][] = 'Database error in line '.__LINE__.': ' . mysqli_error($connid);
 			}
+			// read a temporary set of settings
+			$tempSettingsResult = @mysqli_query($connid, "SELECT name, value FROM `".$db_settings['settings_table']."`";
+			if ($tempSettingsResult !== false) {
+				$update['errors'][] = 'Database error in line '.__LINE__.': ' . mysqli_error($connid);
+			} else {
+				// store the temporary set in an array
+				// a second row with an existing name will overwrite the first row
+				while ($row = mysqli_fetch_assoc($tempSettingsResult)) {
+					$tempSettings[$row['name']] = $row['value'];
+				}
+				$queryAlterSettingsTable = "TRUNCATE TABLE `".$db_settings['settings_table']."`; ALTER TABLE `".$db_settings['settings_table']."` ENGINE = InnoDB, ADD PRIMARY KEY (`name`);";
+				// empty the settings table and alter it's structure
+				if(!@mysqli_query($connid, $queryAlterSettingsTable)) {
```

Comment:
Does `mysqli_query` works? I don't test it but there is a function called `mysqli_multi_query` for multiple queries.